src/core/components/libp2p.js

```diff
@@ -21,7 +21,9 @@ module.exports = function libp2p (self, config) {
 
   const putAndDial = peerInfo => {
     peerBook.put(peerInfo)
```

Comment:
```suggestion
    peerInfo = peerBook.put(peerInfo)
```

When we discover a peer, we may not have the latest information from our PeerBook, such as the connected multiaddress. Doing `peerBook.put` handles merging this information and the new PeerInfo instance is returned. Updating the peerInfo instance, before performing a check, should make sure we have the correct information.